src/erlcloud_ddb2.erl

```diff
@@ -838,6 +834,65 @@ conditional_op_opt() ->
 expected_opt() ->
     {expected, <<"Expected">>, fun dynamize_expected/1}.
 
+-spec filter_expression_opt() -> opt_table_entry().
+
+filter_expression_opt() ->
+    {filter_expression, <<"FilterExpression">>, fun dynamize_expression/1}.
+
+% This matches the Java API, which asks the user to write their own expressions.
+
+-spec dynamize_expression(expression()) -> binary().
+dynamize_expression(Expression) when is_binary(Expression) ->
+    Expression;
+dynamize_expression(Expression) when is_list(Expression) ->
+    list_to_binary(Expression);
```

Comment:
Oh, we'd better not to add list format neither, for now. See [comment](https://github.com/gleber/erlcloud/issues/209#issuecomment-142305659) in #209